spec/models/miq_report/search_spec.rb

```diff
@@ -45,4 +45,32 @@
 
     end
   end
+
+  context "paged_view_search" do
+    it "should not call get_cached_page to load cached results if target class does not respond to id" do
+      report = MiqReport.new(
+          :name => "VmdbDatabaseSetting",
+          :title => "VmdbDatabaseSetting",
+          :db => "VmdbDatabaseSetting",
+          :cols       => ["name", "description", "value", "minimum_value", "maximum_value", "unit"],
+          :col_order  => ["name", "description", "value", "minimum_value", "maximum_value", "unit"],
+          :headers    => ["Name", "Description", "Value", "Minimum", "Maximum", "Unit"],
+          :order      => "Ascending",
+          :sortby     => ["name"],
+          :group      => "n"
+      )
+      options = {
+          :per_page    =>20,
+          :page        =>1,
+          :targets_hash=>true,
+          :userid      =>"admin"
+      }
+
+      expect(report).to_not receive(:get_cached_page)
+
+      results = attrs = nil
+      Proc.new { results, attrs = report.paged_view_search(options) }.call.should_not raise_error
```

Comment:
this expectation seems hard to set up. do you think it's necessary? or could it be implied in line 73?